services/storage/tests/unit/modules/celery/test_celery.py

```diff
@@ -0,0 +1,125 @@
+import logging
+import time
+from collections.abc import Callable
+from random import randint
+
+import pytest
+from celery import Celery, Task
+from celery.contrib.abortable import AbortableTask
+from simcore_service_storage.main import CeleryTaskQueueClient
+from simcore_service_storage.modules.celery.example_tasks import sync_archive
+from simcore_service_storage.modules.celery.models import TaskContext
+from tenacity import Retrying, retry_if_exception_type, stop_after_delay, wait_fixed
+
+_logger = logging.getLogger(__name__)
+
+
+def failure_task(task: Task) -> str:
+    msg = "my error here"
+    raise ValueError(msg)
+
+
+def dreamer_task(task: AbortableTask) -> list[int]:
+    numbers = []
+    for _ in range(30):
+        if task.is_aborted():
+            _logger.warning("Alarm clock")
+            return numbers
+        numbers.append(randint(1, 90))
+        time.sleep(1)
+    return numbers
+
+
+@pytest.fixture
+def register_celery_tasks() -> Callable[[Celery], None]:
+    def _(celery_app: Celery) -> None:
+        celery_app.task(name="sync_archive", bind=True)(sync_archive)
+        celery_app.task(name="failure_task", bind=True)(failure_task)
+        celery_app.task(name="dreamer_task", base=AbortableTask, bind=True)(
+            dreamer_task
+        )
+
+    return _
+
+
+@pytest.fixture
+def task_context() -> TaskContext:
+    return TaskContext(user_id=1, product_name="test")
+
+
+@pytest.mark.usefixtures("celery_client_app", "celery_worker_app")
```

Comment:
aha! nice! I did not know this existed. We used to have something similar with a library but because a nightmare to maintain